src/backend/metal/src/device.rs

```diff
@@ -512,7 +512,7 @@ impl core::Device<Backend> for Device {
             let base_ptr = buf.0.contents() as *mut u8;
 
             if base_ptr.is_null() {
-                return Err(mapping::Error::InvalidAccess);
+                return Err(unimplemented!());
```

Comment:
why `unimplemented`?

Reply:
We had removed `core::mapping::Error::InvalidAccess` because it was originally for render, but it seems it came back with another meaning. Will put it back.